File: modules/angular2/src/compiler/compiler.ts
export {PLATFORM_DIRECTIVES, PLATFORM_PIPES} from 'angular2/src/core/platform_directives_and_pipes';
export * from 'angular2/src/compiler/template_ast';
export {TEMPLATE_TRANSFORMS} from 'angular2/src/compiler/template_parser';
export {CompilerConfig, RenderTypes} from './config';
export * from './compile_metadata';
export * from './offline_compiler';
export {RuntimeCompiler} from './runtime_compiler';
export * from 'angular2/src/compiler/url_resolver';
export * from 'angular2/src/compiler/xhr';

export {ViewResolver} from './view_resolver';
export {DirectiveResolver} from './directive_resolver';
export {PipeResolver} from './pipe_resolver';

import {assertionsEnabled, Type} from 'angular2/src/facade/lang';
import {TemplateParser} from 'angular2/src/compiler/template_parser';
import {HtmlParser} from 'angular2/src/compiler/html_parser';
import {DirectiveNormalizer} from 'angular2/src/compiler/directive_normalizer';
import {CompileMetadataResolver} from 'angular2/src/compiler/metadata_resolver';
import {StyleCompiler} from 'angular2/src/compiler/style_compiler';
import {ViewCompiler} from 'angular2/src/compiler/view_compiler/view_compiler';
import {CompilerConfig} from './config';
import {ComponentResolver} from 'angular2/src/core/linker/component_resolver';
import {RuntimeCompiler} from 'angular2/src/compiler/runtime_compiler';
import {ElementSchemaRegistry} from 'angular2/src/compiler/schema/element_schema_registry';
import {DomElementSchemaRegistry} from 'angular2/src/compiler/schema/dom_element_schema_registry';
import {UrlResolver, DEFAULT_PACKAGE_URL_PROVIDER} from 'angular2/src/compiler/url_resolver';
import {Parser} from './expression_parser/parser';
import {Lexer} from './expression_parser/lexer';
import {ViewResolver} from './view_resolver';
import {DirectiveResolver} from './directive_resolver';
import {PipeResolver} from './pipe_resolver';

function _createCompilerConfig() {
  return new CompilerConfig(assertionsEnabled(), false, true);
}

/**
 * A set of providers that provide `RuntimeCompiler` and its dependencies to use for
 * template compilation.
 */
export const COMPILER_PROVIDERS: Array<any | Type | {[k: string]: any} | any[]> = /*@ts2dart_const*/ [
  Lexer,
  Parser,
  HtmlParser,
  TemplateParser,
  DirectiveNormalizer,
  CompileMetadataResolver,
  DEFAULT_PACKAGE_URL_PROVIDER,
  StyleCompiler,
  ViewCompiler,
  /*@ts2dart_Provider*/{provide: CompilerConfig, useFactory: _createCompilerConfig, deps: []},
  RuntimeCompiler,
  /*@ts2dart_Provider*/{provide: ComponentResolver, useExisting: RuntimeCompiler},
  DomElementSchemaRegistry,
  /*@ts2dart_Provider*/{provide: ElementSchemaRegistry, useExisting: DomElementSchemaRegistry},
  UrlResolver,
  ViewResolver,
  DirectiveResolver,
  PipeResolver
];

File: lib/src/compiler/compiler.dart
library angular2.src.compiler.compiler;

import "runtime_compiler.dart" show RuntimeCompiler_;
export "template_compiler.dart" show TemplateCompiler;
export "directive_metadata.dart"
    show CompileDirectiveMetadata, CompileTypeMetadata, CompileTemplateMetadata;
export "source_module.dart" show SourceModule, SourceWithImports;
export "package:angular2/src/core/platform_directives_and_pipes.dart"
    show PLATFORM_DIRECTIVES, PLATFORM_PIPES;
import "package:angular2/src/facade/lang.dart" show assertionsEnabled, Type;
import "package:angular2/src/core/di.dart" show provide, Provider;
import "package:angular2/src/compiler/template_parser.dart" show TemplateParser;
import "package:angular2/src/compiler/html_parser.dart" show HtmlParser;
import "package:angular2/src/compiler/template_normalizer.dart"
    show TemplateNormalizer;
import "package:angular2/src/compiler/runtime_metadata.dart"
    show RuntimeMetadataResolver;
import "package:angular2/src/compiler/change_detector_compiler.dart"
    show ChangeDetectionCompiler;
import "package:angular2/src/compiler/style_compiler.dart" show StyleCompiler;
import "package:angular2/src/compiler/command_compiler.dart"
    show CommandCompiler;
import "package:angular2/src/compiler/template_compiler.dart"
    show TemplateCompiler;
import "package:angular2/src/core/change_detection/change_detection.dart"
    show ChangeDetectorGenConfig;
import "package:angular2/src/core/linker/compiler.dart" show Compiler;
import "package:angular2/src/compiler/runtime_compiler.dart"
    show RuntimeCompiler;
import "package:angular2/src/compiler/schema/element_schema_registry.dart"
    show ElementSchemaRegistry;
import "package:angular2/src/compiler/schema/dom_element_schema_registry.dart"
    show DomElementSchemaRegistry;
import "package:angular2/src/compiler/url_resolver.dart" show UrlResolver;
import "package:angular2/src/compiler/app_root_url.dart" show AppRootUrl;
import "package:angular2/src/compiler/anchor_based_app_root_url.dart"
    show AnchorBasedAppRootUrl;
import "package:angular2/src/core/change_detection/change_detection.dart"
    show Parser, Lexer;

_createChangeDetectorGenConfig() {
  return new ChangeDetectorGenConfig(assertionsEnabled(), false, true);
}

const List<
        dynamic /* Type | Provider | List < dynamic > */ > COMPILER_PROVIDERS =
    const [
  Lexer,
  Parser,
  HtmlParser,
  TemplateParser,
  TemplateNormalizer,
  RuntimeMetadataResolver,
  StyleCompiler,
  CommandCompiler,
  ChangeDetectionCompiler,
  const Provider(ChangeDetectorGenConfig,
      useFactory: _createChangeDetectorGenConfig, deps: const []),
  TemplateCompiler,
  const Provider(RuntimeCompiler, useClass: RuntimeCompiler_),
  const Provider(Compiler, useExisting: RuntimeCompiler),
  DomElementSchemaRegistry,
  const Provider(ElementSchemaRegistry, useExisting: DomElementSchemaRegistry),
  AnchorBasedAppRootUrl,
  const Provider(AppRootUrl, useExisting: AnchorBasedAppRootUrl),
  UrlResolver
];
